cloud_functions/main.py

```diff
@@ -36,8 +36,13 @@ def clean_and_upload_window(event, context):
     )
 
     window, window_metadata = window_handler.get_window()
-    cleaned_window = window_handler.clean_window(window, window_metadata)
-    window_handler.persist_window(cleaned_window, window_metadata)
+
+    # Correct timestamps with the sensor node time offset
+    for sensor in window["sensor_data"].keys():
+        for row in window["sensor_data"][sensor]:
+            row[0] += window["sensor_time_offset"]
```

Comment:
Might this be better off done at the data source down in the gateway?

Reply:
Logically, it would seem so, but, because of the way sensor_time_offset is computed and stored (it is just a hack for now, until we have "absolute time" on the sensor node), it is more like metadata for the whole window.

It is easier to apply this operation once the window is fully formed and persisted. And once the sensor node will have unix time synchronization, we can easily cut this out.